### PR TITLE
Re-enable dualuse workload cert generation (CommonName identical to the SAN)

### PR DESCRIPTION
**Pull request by boeboe**
**Please provide a description of this PR:**

Check feature request https://github.com/istio/istio/issues/42114

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [x] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
Per discussion in https://github.com/istio/istio/issues/42114, lets get broader agreement from WG before proceeding with this

File: pilot/cmd/pilot-agent/options/options.go [hunk 1]
```diff
@@ -78,6 +78,8 @@ var (
 		"The grace period ratio for the cert rotation, by default 0.5.").Get()
 	workloadRSAKeySizeEnv = env.Register("WORKLOAD_RSA_KEY_SIZE", 2048,
 		"Specify the RSA key size to use for workload certificates.").Get()
+	isDualUseEnv = env.Register("IS_DUAL_USE", false,
```
Comment (costinm) [inline pilot/cmd/pilot-agent/options/options.go:81]: I don't think this should be a setting in agent - the CA can decide what to include ( and should filter anything else )

File: pkg/security/security.go [hunk 2]
```diff
@@ -163,6 +163,10 @@ type Options struct {
 	// WorkloadRSAKeySize is the size of a private key for a workload certificate.
 	WorkloadRSAKeySize int
 
+	// Whether to enable dual-use mode, which generates a workload certificate with a CommonName identical to
```
Comment (costinm) [inline pkg/security/security.go:166]: Same - what is added to cert should be a CA policy, not something the pods decide.

**Review by costinm — COMMENTED** [review]
I generally think it is a great idea - but:
- should be a CA setting. We can't sign whatever the agent decides to put in the CSR, it must be verified. Also the info doesn't have to be in the CSR - a CA can add or set any field.
- we need to have a clear doc/spec on what we want to set - for example for CN I think we should follow the same conventions as K8S, there is no value in setting a spiffee URL as CN IMO
- we keep discussing adding other fields to the cert - like the Pod ID (can be validated by Istiod - avoids sending it as metadata) and even the canonical service name as a DNS record, which would further increase compat with non-Istio workloads.

**Pull request closed without merging**